### PR TITLE
modules: trusted-firmware-m: Post upmerge fixes

### LINKED ISSUE #54509: Zephyr does not configure TF-M correctly for Hard-Float
**Describe the bug**
Zephyr does not configure the correct Floating Point options in TF-M.
FP design in Armv8.0-M architecture requires consistent FP ABI types
between SPE and NSPE.

**To Reproduce**
N/A

**Expected behavior**
Zephyr needs to set TF-M floating point option to match itself.

**Impact**
Zephyr cannot be enabled for Hard-float and TF-M enabled.

**Logs and console output**

**Environment (please complete the following information):**

**Additional context**
This is a regression after the TF-M 1.7.0 update which renamed the Hard Float options.

### PR DESCRIPTION
**Pull request by joerchan**
Fix setting of TF-M floating point options when floating point is
enabled in the application.
FP design in Armv8.0-M architecture requires consistent FP ABI types
between SPE and NSPE.

Remove setting of the TFM_LIB_MODEL option for IPC and SFN models.
This option is removed together with the library model.

The TF-M platform partition has now been ported to PSA firmware
framework 1.1 and can now be used together with the SFN model.

Fixes #54509

### CODE REVIEW

**Review by microbuilder — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Comment by microbuilder** [comment]
@carlescufi This should be merged in for 3.3 if you have time to have a look.

**Comment by carlescufi** [comment]
> @carlescufi This should be merged in for 3.3 if you have time to have a look.

I am not a release manager for this release. Instead, @stephanosio and @laurenmurphyx64 should decide if it should go in.

**Review by carlescufi — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**
